CMakeLists.txt

```diff
@@ -81,9 +81,9 @@ include(${COGUTIL_DATA_DIR}/cmake/Summary.cmake)
 IF (CMAKE_COMPILER_IS_GNUCXX)
 	# gcc before version 4.8 does not support the c++11 definition
 	# of thread_local.
```

Comment:
Changing that message would be welcome as well. Beside the desire to support Ubuntu xenial https://github.com/opencog/atomspace/issues/1916#issuecomment-444601566, I think @Dagiopia pinpointed the problem https://github.com/opencog/atomspace/issues/1916#issuecomment-443631710

Anybody knows what would an adequate message to justify 5.4 (I haven't followed this very closely)?

Reply:
As discused earlier the gcc with version 4.9 supports only primitive data types while we should use  objects. So, that is better to upgrade the library instead of configuring functions which accept objects. The best gcc version is 5.4 because it is used on ubuntu xenial(16.4) and by most containers.

Reply:
the change in g++ that resulted in our upgrade is this "G++ now allows passing a non-trivially-copyable class via C varargs, which is conditionally-supported with implementation-defined semantics in the standard. This uses the same calling convention as a normal value parameter." from https://www.gnu.org/software/gcc/gcc-5/changes.html

@ngeiswei maybe we should have a quote from the above next to "GCC version must be at least 5.4.0!"  to make it clear to the user.

Reply:
Actually 5.0

Reply:
Thanks @Dagiopia , so yes I suppose we should set the minimum version to 5.0 with a comment explaining about gcc support for passing non-trivially copyable class via C varargs.